include/pf-applications/lac/dynamic_block_vector.h

```diff
@@ -70,22 +68,17 @@ namespace dealii
         void
         reinit(const unsigned int n, const bool omit_zeroing_entries = false)
         {
-          block_counter = n;
-
           const unsigned int old_blocks_size = blocks.size();
 
-          if (n_blocks() > old_blocks_size)
-            {
-              blocks.resize(n_blocks());
+          blocks.resize(n);
 
-              for (unsigned int b = old_blocks_size; b < n_blocks(); ++b)
-                {
-                  if (blocks[b] == nullptr)
-                    blocks[b] = std::make_shared<BlockType>();
+          for (unsigned int b = old_blocks_size; b < n_blocks(); ++b)
```

Comment:
For some reason `old_blocks_size` was not correct (some inconsistency of `block_counter` and `blocks.size()`).